src/System.Management.Automation/utils/ClrFacade.cs

```diff
@@ -58,9 +58,36 @@ internal static IEnumerable<Assembly> GetAssemblies(TypeResolutionState typeReso
         /// </param>
         internal static IEnumerable<Assembly> GetAssemblies(string namespaceQualifiedTypeName = null)
         {
-            return PSAssemblyLoadContext.GetAssembly(namespaceQualifiedTypeName) ??
-                   AssemblyLoadContext.Default.Assemblies.Where(a =>
-                       !a.FullName.StartsWith(TypeDefiner.DynamicClassAssemblyFullNamePrefix, StringComparison.Ordinal));
+            return PSAssemblyLoadContext.GetAssembly(namespaceQualifiedTypeName) ?? GetPSVisibleAssemblies();
+        }
+
+        /// <summary>
+        /// Return assemblies from the default load context and the 'individual' load contexts.
+        /// The 'individual' load contexts are the ones holding assemblies loaded via 'Assembly.Load(byte[])' and 'Assembly.LoadFile'.
+        /// Assemblies loaded in any custom load contexts are not consider visible to PowerShell to avoid type identity issues.
+        /// </summary>
+        private static IEnumerable<Assembly> GetPSVisibleAssemblies()
+        {
+            const string IndividualAssemblyLoadContext = "System.Runtime.Loader.IndividualAssemblyLoadContext";
+
+            foreach (Assembly assembly in AssemblyLoadContext.Default.Assemblies)
+            {
+                if (!assembly.FullName.StartsWith(TypeDefiner.DynamicClassAssemblyFullNamePrefix, StringComparison.Ordinal))
+                {
+                    yield return assembly;
+                }
+            }
+
+            foreach (AssemblyLoadContext context in AssemblyLoadContext.All)
+            {
+                if (IndividualAssemblyLoadContext.Equals(context.GetType().FullName, StringComparison.Ordinal))
```

Comment:
Can we do:
```suggestion
                if (context is IndividualAssemblyLoadContext)
```

Reply:
`is` won't work dynamically, it takes a type metadata reference instead of a `System.Type` object (even in raw IL it isn't possible).  You could do `typeof(int).Assembly.GetType("FullTypeNameHere.IALC").IsAssignableFrom(context.GetType())` but then you depend on the assembly where the non-public type is located staying the same.